### DISCUSSION TITLE
Container "exec format error" Error

### DISCUSSION THREAD

**Discussion by TroyDavidson** [serverless/serverless#11177] [General]
I'm using a container to hold my lambda functions.  I have to do it this way for security reasons.

The project is in PHP and using Bref for the lambda translation layer.  I have serverless build the container and push it to ECR.  I then hit an end point, and I get the following error:

`IMAGE	Launch error: fork/exec EntryPoints/auth.php: exec format error	Entrypoint: [EntryPoints/auth.php]	Cmd: [/lambda-entrypoint.sh]	WorkingDir: [/var/task]`

I've tried with a container built in MacOS, Linux, and also Github using actions.  I've read some pages that said what OS the container was built on would cause this issue.  That's why I've tried on the various platforms.

This is from the serverless.yml:

functions:
  authPost:
    image:
      name: <name>
      command:
        - "/lambda-entrypoint.sh"
      entryPoint:
        - "EntryPoints/auth.php"

Also, I just ran the invoke command and got the following:
`tdavidson $  serverless invoke --function authPost`
`{
    "errorMessage": "RequestId: <id> Error: fork/exec EntryPoints/auth.php: exec format error",
    "errorType": "Runtime.InvalidEntrypoint"
}`

The "InvalidEntrypoint" is interesting.  Should I be doing something else?

Any help would be great.  Thanks.

**Comment by pgrzesik** [comment]
Hey @TroyDavidson, 

looking at the report it's a bit hard to provide guidance so I'll try to ask a few questions:
1. What is the content of `lambda-entrypoint.sh`? What is happening in that script?
2. What is the content of `EntryPoints/auth.php`? Is that your actual Lambda logic? 
 
In general, the entrypoint should be the wrapper that runs your function (it should implement the runtime interface) and `command` is being passed to that entrypoint, so it's usually your actual handler for a function. I'm guessing you might have it backwards in your case?

**Reply by TroyDavidson** [reply]
That was it.  I did have the two files backwards.  Thanks for the help.

**Reply by pgrzesik** [reply]
I'm happy you were able to resolve your issue 🙌
